nim_status/client.nim

```diff
@@ -224,6 +230,36 @@ proc storeImportedWalletAccount(self: StatusObject, privateKey: SkSecretKey,
   except Exception as e:
     return WalletAccountResult.err e.msg
 
+proc addCustomToken*(self: StatusObject, address, name, symbol, color, decimals: string): AddCustomTokenResult =
+  if not self.isLoggedIn:
+    return AddCustomTokenResult.err "Not logged in. You must be logged in to " &
+      "add a new custom token."
+
+  var uintDecimals: uint
+  if decimals != "":
+    uintDecimals = parseUInt(decimals)
+  # TODO
+  let token = Token(address: cast[Address](address), name: name, symbol: symbol, color: color, decimals: uintDecimals)
```

Comment:
To me, it makes more sense to accept typed arguments (`address: Address` and `decimals: int`) in the `addCustomToken` proc signature as we are operating in the nim-status library which requires these types to function correctly. It should be up to the consuming library to convert to the correct type before calling functions in `nim-status/client`.
```suggestion
  let token = Token(address: address, name: name, symbol: symbol, color: color, decimals: uintDecimals)
```
Note also, that types should be *type converted* instead of *casted* where possible, ie `Address(address)` instead of `cast[Address](address)` as a cast is not needed in the majority of cases and could cause unwanted side effects if not used correctly.

Reply:
Removed casts, thanks for pointing this out.